services/web/server/src/simcore_service_webserver/catalog_client.py

```diff
@@ -0,0 +1,123 @@
+""" Requests to catalog service API
+
+"""
+import logging
+import urllib.parse
+from typing import Any, Dict, List, Optional, Union
+
+from aiohttp import ContentTypeError, web
+from servicelib.rest_responses import wrap_as_envelope
+from yarl import URL
+
+from ._meta import api_version_prefix
+from .catalog_config import KCATALOG_ORIGIN, KCATALOG_VERSION_PREFIX, get_client_session
+from .constants import X_PRODUCT_NAME_HEADER
+
+logger = logging.getLogger(__name__)
+
+
+async def is_service_responsive(app: web.Application):
+    """ Returns true if catalog is ready """
+    origin: Optional[URL] = app.get(KCATALOG_ORIGIN)
+
+    if not origin:  # service was not enabled!
+        return False
+
+    client = get_client_session(app)
+
+    # call to health-check entry-point
+    async with client.get(origin, ssl=False) as resp:
+        return resp.status == 200
+
+
+def to_backend_service(rel_url: URL, origin: URL, version_prefix: str) -> URL:
+    """Translates relative url to backend catalog service url
+
+    E.g. https://osparc.io/v0/catalog/dags -> http://catalog:8080/v0/dags
+    """
+    assert not rel_url.is_absolute()  # nosec
+    new_path = rel_url.path.replace(
+        f"/{api_version_prefix}/catalog", f"/{version_prefix}"
+    )
+    return origin.with_path(new_path).with_query(rel_url.query)
+
+
+async def make_request_and_envelope_response(
+    app: web.Application,
+    method: str,
+    url: URL,
+    headers: Optional[Dict[str, str]] = None,
+    data: Optional[bytes] = None,
+) -> web.Response:
+    """
+    Helper to forward a request to the catalog service
+    """
+    session = get_client_session(app)
+
+    try:
+        async with session.request(method, url, headers=headers, data=data) as resp:
+
+            is_error = resp.status >= 400
+            # catalog backend sometimes sends error in plan=in text
+            payload: Union[Dict, str] = {}
+            try:
+                payload = await resp.json()
+            except ContentTypeError:
```

Comment:
we cannot force catalog to always return json? like api-server and director-v2?

Reply:
Fixed both in the catalog service and here